product_configurator/models/product_attribute.py

```diff
@@ -0,0 +1,445 @@
+        default.update({"name": self.name + " (copy)"})
+        product = super().copy(default)
+        return product
+
+    active = fields.Boolean(
+        default=True,
+        help="By unchecking the active field you can "
+        "disable a attribute value without deleting it",
+    )
+    product_id = fields.Many2one(comodel_name="product.product")
+    image = fields.Binary(
+        attachment=True,
+        help="Attribute value image (Display on website for radio buttons)",
+    )
+
+    @api.model
+    def get_attribute_value_extra_prices(
+        self, product_tmpl_id, pt_attr_value_ids, pricelist=None
+    ):
+        extra_prices = {}
+        if not pricelist:
+            pricelist = self.env.user.partner_id.property_product_pricelist
+
+        related_product_av_ids = self.env["product.attribute.value"].search(
+            [("id", "in", pt_attr_value_ids.ids), ("product_id", "!=", False)]
+        )
+        extra_prices = {
+            av.id: av.product_id.with_context(
+                pricelist=pricelist.id
+            )._get_contextual_price()
+            for av in related_product_av_ids
+        }
+        remaining_av_ids = pt_attr_value_ids - related_product_av_ids
+        pe_lines = self.env["product.template.attribute.value"].search(
+            [
+                ("product_attribute_value_id", "in", remaining_av_ids.ids),
+                ("product_tmpl_id", "=", product_tmpl_id),
+            ]
+        )
+        for line in pe_lines:
+            attr_val_id = line.product_attribute_value_id
+            if attr_val_id.id not in extra_prices:
+                extra_prices[attr_val_id.id] = 0
+            extra_prices[attr_val_id.id] += line.price_extra
+        return extra_prices
+
+    def _compute_display_name(self):
+        # useless return to make pylint happy
+        res = super()._compute_display_name()
+        if not self.env.context.get("show_price_extra"):
+            return res
+        product_template_id = self.env.context.get("active_id", False)
+        price_precision = self.env["decimal.precision"].precision_get("Product Price")
+        for attribute in self:
+            extra_prices = attribute.get_attribute_value_extra_prices(
+                product_tmpl_id=product_template_id, pt_attr_value_ids=attribute
+            )
+            price_extra = extra_prices.get(attribute.id)
+            if price_extra:
+                name = f"{attribute.name} ( +{price_extra:.{price_precision}f} )"
+                attribute.display_name = name
+
+    @api.model
+    def name_search(self, name="", args=None, operator="ilike", limit=100):
+        """Use name_search as a domain restriction for the frontend to show
+        only values set on the product template taking all the configuration
+        restrictions into account.
+
+        TODO: This only works when activating the selection not when typing
+        """
+        product_tmpl_id = self.env.context.get("_cfg_product_tmpl_id")
```

Comment:
Can we take the opportunity to write unit tests for this method? 🙏🏻

Reply:
We take care of the test cases also in this MR